test-utils/testlib/src/fees_utils.rs

```diff
@@ -19,49 +19,69 @@ impl FeeHelper {
     }
 
     pub fn gas_to_balance_inflated(&self, gas: Gas) -> Balance {
-        gas as Balance
+        gas.as_gas() as Balance
             * (self.gas_price * (*self.cfg().pessimistic_gas_price_inflation_ratio.numer() as u128)
                 / (*self.cfg().pessimistic_gas_price_inflation_ratio.denom() as u128))
     }
 
     pub fn gas_to_balance(&self, gas: Gas) -> Balance {
-        gas as Balance * self.gas_price
+        gas.as_gas() as Balance * self.gas_price
     }
 
     pub fn gas_burnt_to_reward(&self, gas_burnt: Gas) -> Balance {
-        let gas_reward = gas_burnt * *self.cfg().burnt_gas_reward.numer() as u64
-            / *self.cfg().burnt_gas_reward.denom() as u64;
+        let gas_reward = Gas::from_gas(
+            gas_burnt.as_gas() * *self.cfg().burnt_gas_reward.numer() as u64
+                / *self.cfg().burnt_gas_reward.denom() as u64,
+        );
         self.gas_to_balance(gas_reward)
     }
 
     pub fn create_account_cost(&self) -> Balance {
-        let exec_gas = self.cfg().fee(ActionCosts::new_action_receipt).exec_fee()
-            + self.cfg().fee(ActionCosts::create_account).exec_fee();
-        let send_gas = self.cfg().fee(ActionCosts::new_action_receipt).send_fee(false)
-            + self.cfg().fee(ActionCosts::create_account).send_fee(false);
-        self.gas_to_balance(exec_gas + send_gas)
+        let exec_gas = self
+            .cfg()
+            .fee(ActionCosts::new_action_receipt)
+            .exec_fee()
+            .saturating_add(self.cfg().fee(ActionCosts::create_account).exec_fee());
+        let send_gas = self
+            .cfg()
+            .fee(ActionCosts::new_action_receipt)
+            .send_fee(false)
+            .saturating_add(self.cfg().fee(ActionCosts::create_account).send_fee(false));
+        self.gas_to_balance(exec_gas.saturating_add(send_gas))
     }
 
     pub fn create_account_transfer_full_key_fee(&self) -> Gas {
-        let exec_gas = self.cfg().fee(ActionCosts::new_action_receipt).exec_fee()
-            + self.cfg().fee(ActionCosts::create_account).exec_fee()
-            + self.cfg().fee(ActionCosts::transfer).exec_fee()
-            + self.cfg().fee(ActionCosts::add_full_access_key).exec_fee();
-        let send_gas = self.cfg().fee(ActionCosts::new_action_receipt).send_fee(false)
-            + self.cfg().fee(ActionCosts::create_account).send_fee(false)
-            + self.cfg().fee(ActionCosts::transfer).send_fee(false)
-            + self.cfg().fee(ActionCosts::add_full_access_key).send_fee(false);
-        exec_gas + send_gas
+        let exec_gas = self
+            .cfg()
+            .fee(ActionCosts::new_action_receipt)
+            .exec_fee()
+            .saturating_add(self.cfg().fee(ActionCosts::create_account).exec_fee())
+            .saturating_add(self.cfg().fee(ActionCosts::transfer).exec_fee())
+            .saturating_add(self.cfg().fee(ActionCosts::add_full_access_key).exec_fee());
```

Comment:
It is really hard to review. Let's extract the individual fees into separate variables and then `saturating_add` them